File: src/DefScript.pas
unit DefScript;


interface

const
    // Sequence ID.
  SEQ_STOP      = 0;
  SEQ_KEY_START = 1;
  SEQ_KEY_STOP  = 2;
  SEQ_KEY_1     = 3;
  SEQ_KEY_2     = 4;
  SEQ_KEY_3     = 5;
  SEQ_KEY_4     = 6;
  SEQ_KEY_5     = 7;
  SEQ_KEY_6     = 8;
  SEQ_KEY_7     = 9;
  SEQ_KEY_8     = 10;
  SEQ_KEY_9     = 11;
  SEQ_KEY_SCAN  = 12;
  SEQ_CAM_ZONE  = 13;
  SEQ_UNLOAD_ZONE = 14;

  SEQ_PRE_STOP  = 15;
  SEQ_REPORT    = 16;
  SEQ_1         = 17;
  SEQ_2         = 18;
  SEQ_3         = 19;
  SEQ_4         = 20;
  SEQ_5         = 21;
  SEQ_6         = 22;
  SEQ_7         = 23;
  SEQ_8         = 24;
  SEQ_9         = 25;
  SEQ_Finish    = 26;

  SEQ_MAINT_1   = 31;
  SEQ_MAINT_2   = 32;
  SEQ_MAINT_3   = 33;
  SEQ_MAINT_4   = 34;
  SEQ_MAINT_5   = 35;

  SEQ_MAX       = SEQ_MAINT_5;


  SEQ_ERR_NONE      = 0;
  SEQ_ERR_RUNNING   = 1;

  DAYA_NONE         = 0;
  DATA_TYPE_HEX     = 1;
  DATA_TYPE_DEC     = 2;
  DATA_TYPE_REAL    = 3;
  DATA_TYPE_STR     = 6;
//  DATA_TYPE_MODE1   = 8;

  REF_IDX_NONE      = 0;
  REF_IDX_MAX       = 1;
  REF_IDX_MIN       = 2;
  REF_IDX_DIFF      = 3;
  REF_IDX_AVR       = 4;
  REF_IDX_DIFF_P2P2 = 5;
  REF_IDX_AVR_JITER = 6;
  REF_IDX_SLOPE_ROW = 7;
  REF_IDX_SLOPE_COL = 8;
  REF_IDX_JIT_DELTA = 9;
  REF_IDX_RAWCS_OPEN  = 11;
  REF_IDX_RAWCS_OPEN2 = 12;
//  REF_IDX_CV5A_OPEN   = 15;
  REF_IDX_GET_CAL     = 20; // frame AIEA Average¡ÍiiAC ¨Ï©ª¢®i¡Ë?e AU¡Íi¡Ë? ¢®¨¡e¢®ieEA ¨Ïoo¡§¢®U¡Ë?¡Ë¢ç AuAa.
  REF_IDX_DOWN_FW     = 30;
  REF_IDX_ID_UPDATE   = 31;
//  REF_IDX_Y3_FLASH_WR = 32;
  REF_IDX_RTY_MODIFY  = 300;

  LIMIT_TYPE_NG           = 0;
  LIMIT_TYPE_IS           = 1;
  LIMIT_TYPE_ISNOT        = 2;
  LIMIT_TYPE_MIN          = 3;
  LIMIT_TYPE_MAX          = 4;
  LIMIT_TYPE_MAXMIN       = 5;
  LIMIT_TYPE_MAXMIN_FLOAT = 6;
  LIMIT_TYPE_MAX_SUB_MIN  = 7;
  LIMIT_TYPE_STR          = 8;
  LIMIT_TYPE_LOG          = 9;
  LIMIT_TYPE_FWVER        = 10;
//  LIMIT_TYPE_SPEC_BY_CSV1 = ;
//  LIMIT_TYPE_SPEC_BY_CSV2 = ;
//---------------------------------------------------------
  FUNC_CALL_START         = 'Func_call::';
	func_call_TxIcInit      = 'Func_call::TxIcInit()'; 	  //TxIcInit                 .mpt
	func_call_Module_on     = 'Func_call::ModuleOn()';    //Module On Code.          .mion
	func_call_Module_off    = 'Func_call::ModuleOff()';   //Module Off Code.         .miOff
	func_call_Power_On      = 'Func_call::PwrOn()';       //Power On Full Init Code. .pwon.
	func_call_Power_Off     = 'Func_call::PwrOff()';      //Power Off Code.          .pwoff.
  func_call_Power_On_Auto = 'Func_call::PwrOn_Auto()';  // Power On Auto Code      .miau

  func_call_Otp_Write     = 'Func_call::OtpWrite()';    // OTP Write Code.          .otpw
  func_call_Otp_Read      = 'Func_call::OtpRead()';     // Otp Read Code.           .otpr
  func_call_ScreenCode    = 'Func_call::ScreenCode()';  // Screen Code              .misc.

  CODE_TXICINIT      = 1;
  CODE_MODULE_ON     = 2;
  CODE_MODULE_OFF    = 3;
  CODE_POWER_ON      = 4;
  CODE_POWER_OFF     = 5;
  CODE_POWER_ON_AUTO = 6;
  CODE_OTP_WRITE     = 7;
  CODE_OTP_READ      = 8;
  CODE_SCR_CODE      = 9;
  CODE_MAX           = CODE_SCR_CODE;

  CMD_ITEM_1        = 'delay.ms';
  CMD_ITEM_2        = 'mipiic.write';
  CMD_ITEM_3        = 'mipi.write';
  CMD_ITEM_4        = 'vbat.on';
  CMD_ITEM_5        = 'vneg.on';
  CMD_ITEM_6        = 'vext.on';
  CMD_ITEM_7        = 'vcc.on';
  CMD_ITEM_8        = 'vlcd.on';
  CMD_ITEM_9        = 'mdm.reset';
  CMD_ITEM_10       = 'lcm.reset';
  CMD_ITEM_11       = 'gpio.on';
  CMD_ITEM_12       = 'gpio.off';
  CMD_ITEM_13       = 'mdm.init';
  CMD_ITEM_14       = 'mipi.on';
  CMD_ITEM_15       = 'vbat.off';
  CMD_ITEM_16       = 'vneg.off';
  CMD_ITEM_17       = 'vext.off';
  CMD_ITEM_18       = 'vcc.off';
  CMD_ITEM_19       = 'vlcd.off';
  CMD_ITEM_20       = 'mipi.read';
  CMD_ITEM_21       = 'mipiic.read';
  CMD_ITEM_22       = 'vel.on';
  CMD_ITEM_23       = 'vel.off';
  CMD_ITEM_24       = 'mdm.deinit';

  CMD_ITEM_25       = 'vci.on';
  CMD_ITEM_26       = 'dvdd.on';
  CMD_ITEM_27       = 'vdd.on';
  CMD_ITEM_28       = 'vpp.on';
  CMD_ITEM_29       = 'vneg.on';

  CMD_ITEM_30       = 'gpio.on';
  CMD_ITEM_31       = 'gpio.off';
  CMD_ITEM_32       = 'mipi.on';
  CMD_ITEM_33       = 'vci.off';
  CMD_ITEM_34       = 'dvdd.off';
  CMD_ITEM_35       = 'vdd.off';
  CMD_ITEM_36       = 'vpp.off';
  CMD_ITEM_37       = 'mipi.off';
  CMD_ITEM_38       = 'touch.model';
  CMD_ITEM_39       = 'mipi.wpacket';
  CMD_ITEM_40       = 'voltage.limit';

  // for PG To Comm.
  // Pascal Script SIG ID.
  PP_SIGID_1        = 1;    // Power
  PP_SIGID_2        = 2;    // Pattern Display
  PP_SIGID_3        = 3;

  // Pascal Script Command - 4th param.
  PP_COMMAD_PWR_OFF = 0;
  PP_COMMAD_PWR_ON  = 1;
  PP_COMMAD_PWR_ON_AUTOCODE  = 2;
  PP_COMMAD_PWR_OFF_RESET = 3;
  PP_COMMAD_PWR_ON_RESET  = 4;

  PP_COMMAD_MES_OFF = 0;
  PP_COMMAD_MES_ON  = 1;
  PP_COMMAD_PAT_GRP = 0;
  PP_COMMAD_PAT_SNG = 1;

  end_func          = '}_end_func';

  ERR_ST_NONE       = 0;
  ERR_ST_SEME       = 1;
implementation

end.

File: src_X3584_CSharp/ITOLED.Core/Definitions/DefScript.cs
// ---------------------------------------------------------------------------
// DefScript.cs
// Converted from Delphi: DefScript.pas
// Sequence IDs, data types, reference indices, limit types,
// function call strings, PG command items, and Pascal Script constants.
// ---------------------------------------------------------------------------

namespace Dongaeltek.ITOLED.Core.Definitions;

/// <summary>
/// Script definition constants used for sequence control, data types,
/// reference indices, limit types, function call identifiers,
/// PG command items, and Pascal Script communication.
/// <para>Delphi source: DefScript.pas</para>
/// </summary>
public static class DefScript
{
    #region Sequence IDs (SEQ_*)

    /// <summary>Sequence stop.</summary>
    public const int SeqStop = 0;

    /// <summary>Key start sequence.</summary>
    public const int SeqKeyStart = 1;

    /// <summary>Key stop sequence.</summary>
    public const int SeqKeyStop = 2;

    public const int SeqKey1 = 3;
    public const int SeqKey2 = 4;
    public const int SeqKey3 = 5;
    public const int SeqKey4 = 6;
    public const int SeqKey5 = 7;
    public const int SeqKey6 = 8;
    public const int SeqKey7 = 9;
    public const int SeqKey8 = 10;
    public const int SeqKey9 = 11;

    /// <summary>Key scan sequence.</summary>
    public const int SeqKeyScan = 12;

    /// <summary>Camera zone sequence.</summary>
    public const int SeqCamZone = 13;

    /// <summary>Unload zone sequence.</summary>
    public const int SeqUnloadZone = 14;

    /// <summary>Pre-stop sequence.</summary>
    public const int SeqPreStop = 15;

    /// <summary>Report sequence.</summary>
    public const int SeqReport = 16;

    public const int Seq1 = 17;
    public const int Seq2 = 18;
    public const int Seq3 = 19;
    public const int Seq4 = 20;
    public const int Seq5 = 21;
    public const int Seq6 = 22;
    public const int Seq7 = 23;
    public const int Seq8 = 24;
    public const int Seq9 = 25;

    /// <summary>Sequence finish. Delphi: SEQ_Finish</summary>
    public const int SeqFinish = 26;

    public const int SeqRestart1 = 27;
    public const int SeqRestart2 = 28;
    public const int SeqRestart3 = 29;

    // Note: 30 is intentionally skipped in the original Delphi source.

    public const int SeqMaint1 = 31;
    public const int SeqMaint2 = 32;
    public const int SeqMaint3 = 33;
    public const int SeqMaint4 = 34;
    public const int SeqMaint5 = 35;

    /// <summary>Maximum sequence ID (= SeqMaint5). Delphi: SEQ_MAX</summary>
    public const int SeqMax = SeqMaint5;

    #endregion

    #region Sequence Error Codes (SEQ_ERR_*)

    public const int SeqErrNone = 0;
    public const int SeqErrRunning = 1;

    #endregion

    #region Data Types (DATA_TYPE_*)

    /// <summary>Delphi: DAYA_NONE (original typo preserved in value).</summary>
    public const int DataNone = 0;

    /// <summary>Hexadecimal data type.</summary>
    public const int DataTypeHex = 1;

    /// <summary>Decimal data type.</summary>
    public const int DataTypeDec = 2;

    /// <summary>Real (floating-point) data type.</summary>
    public const int DataTypeReal = 3;

    /// <summary>String data type.</summary>
    public const int DataTypeStr = 6;

    #endregion

    #region Reference Indices (REF_IDX_*)

    public const int RefIdxNone = 0;
    public const int RefIdxMax = 1;
    public const int RefIdxMin = 2;
    public const int RefIdxDiff = 3;
    public const int RefIdxAvr = 4;

    /// <summary>Peak-to-peak difference. Delphi: REF_IDX_DIFF_P2P2</summary>
    public const int RefIdxDiffP2P2 = 5;

    /// <summary>Average jitter. Delphi: REF_IDX_AVR_JITER</summary>
    public const int RefIdxAvrJiter = 6;

    /// <summary>Slope row. Delphi: REF_IDX_SLOPE_ROW</summary>
    public const int RefIdxSlopeRow = 7;

    /// <summary>Slope column. Delphi: REF_IDX_SLOPE_COL</summary>
    public const int RefIdxSlopeCol = 8;

    /// <summary>Jitter delta. Delphi: REF_IDX_JIT_DELTA</summary>
    public const int RefIdxJitDelta = 9;

    /// <summary>Raw CS open. Delphi: REF_IDX_RAWCS_OPEN</summary>
    public const int RefIdxRawcsOpen = 11;

    /// <summary>Raw CS open 2. Delphi: REF_IDX_RAWCS_OPEN2</summary>
    public const int RefIdxRawcsOpen2 = 12;

    /// <summary>Get calibration. Delphi: REF_IDX_GET_CAL</summary>
    public const int RefIdxGetCal = 20;

    /// <summary>Download firmware. Delphi: REF_IDX_DOWN_FW</summary>
    public const int RefIdxDownFw = 30;

    /// <summary>ID update. Delphi: REF_IDX_ID_UPDATE</summary>
    public const int RefIdxIdUpdate = 31;

    /// <summary>Retry modify. Delphi: REF_IDX_RTY_MODIFY</summary>
    public const int RefIdxRtyModify = 300;

    #endregion

    #region Limit Types (LIMIT_TYPE_*)

    public const int LimitTypeNg = 0;
    public const int LimitTypeIs = 1;
    public const int LimitTypeIsNot = 2;
    public const int LimitTypeMin = 3;
    public const int LimitTypeMax = 4;
    public const int LimitTypeMaxMin = 5;
    public const int LimitTypeMaxMinFloat = 6;
    public const int LimitTypeMaxSubMin = 7;
    public const int LimitTypeStr = 8;
    public const int LimitTypeLog = 9;
    public const int LimitTypeFwVer = 10;

    #endregion

    #region Function Call Strings (func_call_*)

    /// <summary>Function call prefix. Delphi: FUNC_CALL_START</summary>
    public const string FuncCallStart = "Func_call::";

    /// <summary>TxIcInit (.mpt). Delphi: func_call_TxIcInit</summary>
    public const string FuncCallTxIcInit = "Func_call::TxIcInit()";

    /// <summary>Module On (.mion). Delphi: func_call_Module_on</summary>
    public const string FuncCallModuleOn = "Func_call::ModuleOn()";

    /// <summary>Module Off (.miOff). Delphi: func_call_Module_off</summary>
    public const string FuncCallModuleOff = "Func_call::ModuleOff()";

    /// <summary>Power On Full Init (.pwon). Delphi: func_call_Power_On</summary>
    public const string FuncCallPowerOn = "Func_call::PwrOn()";

    /// <summary>Power Off (.pwoff). Delphi: func_call_Power_Off</summary>
    public const string FuncCallPowerOff = "Func_call::PwrOff()";

    /// <summary>Power On Auto (.miau). Delphi: func_call_Power_On_Auto</summary>
    public const string FuncCallPowerOnAuto = "Func_call::PwrOn_Auto()";

    /// <summary>OTP Write (.otpw). Delphi: func_call_Otp_Write</summary>
    public const string FuncCallOtpWrite = "Func_call::OtpWrite()";

    /// <summary>OTP Read (.otpr). Delphi: func_call_Otp_Read</summary>
    public const string FuncCallOtpRead = "Func_call::OtpRead()";

    /// <summary>Screen Code (.misc). Delphi: func_call_ScreenCode</summary>
    public const string FuncCallScreenCode = "Func_call::ScreenCode()";

    #endregion

    #region Code IDs (CODE_*)

    public const int CodeTxIcInit = 1;
    public const int CodeModuleOn = 2;
    public const int CodeModuleOff = 3;
    public const int CodePowerOn = 4;
    public const int CodePowerOff = 5;
    public const int CodePowerOnAuto = 6;
    public const int CodeOtpWrite = 7;
    public const int CodeOtpRead = 8;
    public const int CodeScrCode = 9;

    /// <summary>Maximum code ID (= CodeScrCode). Delphi: CODE_MAX</summary>
    public const int CodeMax = CodeScrCode;

    #endregion

    #region Command Items (CMD_ITEM_*)

    public const string CmdItem1 = "delay.ms";
    public const string CmdItem2 = "mipiic.write";
    public const string CmdItem3 = "mipi.write";
    public const string CmdItem4 = "vbat.on";
    public const string CmdItem5 = "vneg.on";
    public const string CmdItem6 = "vext.on";
    public const string CmdItem7 = "vcc.on";
    public const string CmdItem8 = "vlcd.on";
    public const string CmdItem9 = "mdm.reset";
    public const string CmdItem10 = "lcm.reset";
    public const string CmdItem11 = "gpio.on";
    public const string CmdItem12 = "gpio.off";
    public const string CmdItem13 = "mdm.init";
    public const string CmdItem14 = "mipi.on";
    public const string CmdItem15 = "vbat.off";
    public const string CmdItem16 = "vneg.off";
    public const string CmdItem17 = "vext.off";
    public const string CmdItem18 = "vcc.off";
    public const string CmdItem19 = "vlcd.off";
    public const string CmdItem20 = "mipi.read";
    public const string CmdItem21 = "mipiic.read";
    public const string CmdItem22 = "vel.on";
    public const string CmdItem23 = "vel.off";
    public const string CmdItem24 = "mdm.deinit";
    public const string CmdItem25 = "vci.on";
    public const string CmdItem26 = "dvdd.on";
    public const string CmdItem27 = "vdd.on";
    public const string CmdItem28 = "vpp.on";

    /// <summary>Note: same value as CmdItem5. Delphi: CMD_ITEM_29 = 'vneg.on'</summary>
    public const string CmdItem29 = "vneg.on";

    /// <summary>Note: same value as CmdItem11. Delphi: CMD_ITEM_30 = 'gpio.on'</summary>
    public const string CmdItem30 = "gpio.on";

    /// <summary>Note: same value as CmdItem12. Delphi: CMD_ITEM_31 = 'gpio.off'</summary>
    public const string CmdItem31 = "gpio.off";

    /// <summary>Note: same value as CmdItem14. Delphi: CMD_ITEM_32 = 'mipi.on'</summary>
    public const string CmdItem32 = "mipi.on";

    public const string CmdItem33 = "vci.off";
    public const string CmdItem34 = "dvdd.off";
    public const string CmdItem35 = "vdd.off";
    public const string CmdItem36 = "vpp.off";
    public const string CmdItem37 = "mipi.off";
    public const string CmdItem38 = "touch.model";
    public const string CmdItem39 = "mipi.wpacket";
    public const string CmdItem40 = "voltage.limit";

    #endregion

    #region PG Communication - Pascal Script Signal IDs (PP_SIGID_*)

    /// <summary>Power signal. Delphi: PP_SIGID_1</summary>
    public const int PpSigIdPower = 1;

    /// <summary>Pattern display signal. Delphi: PP_SIGID_2</summary>
    public const int PpSigIdPatternDisplay = 2;

    /// <summary>Signal 3. Delphi: PP_SIGID_3</summary>
    public const int PpSigId3 = 3;

    #endregion

    #region PG Communication - Pascal Script Commands (PP_COMMAD_*)

    /// <summary>Power off command. Delphi: PP_COMMAD_PWR_OFF</summary>
    public const int PpCommandPwrOff = 0;

    /// <summary>Power on command. Delphi: PP_COMMAD_PWR_ON</summary>
    public const int PpCommandPwrOn = 1;

    /// <summary>Power on with auto-code. Delphi: PP_COMMAD_PWR_ON_AUTOCODE</summary>
    public const int PpCommandPwrOnAutoCode = 2;

    /// <summary>Power off with reset. Delphi: PP_COMMAD_PWR_OFF_RESET</summary>
    public const int PpCommandPwrOffReset = 3;

    /// <summary>Power on with reset. Delphi: PP_COMMAD_PWR_ON_RESET</summary>
    public const int PpCommandPwrOnReset = 4;

    /// <summary>Power off VSYS. Delphi: PP_COMMAD_PWR_OFF_VSYS</summary>
    public const int PpCommandPwrOffVsys = 5;

    /// <summary>Power on VSYS. Delphi: PP_COMMAD_PWR_ON_VSYS</summary>
    public const int PpCommandPwrOnVsys = 6;

    /// <summary>Power off VSYS with reset. Delphi: PP_COMMAD_PWR_OFF_VSYS_RESET</summary>
    public const int PpCommandPwrOffVsysReset = 7;

    /// <summary>Power on VSYS with reset. Delphi: PP_COMMAD_PWR_ON_VSYS_RESET</summary>
    public const int PpCommandPwrOnVsysReset = 8;

    /// <summary>Measurement off. Delphi: PP_COMMAD_MES_OFF</summary>
    public const int PpCommandMesOff = 0;

    /// <summary>Measurement on. Delphi: PP_COMMAD_MES_ON</summary>
    public const int PpCommandMesOn = 1;

    /// <summary>Pattern group. Delphi: PP_COMMAD_PAT_GRP</summary>
    public const int PpCommandPatGrp = 0;

    /// <summary>Pattern single. Delphi: PP_COMMAD_PAT_SNG</summary>
    public const int PpCommandPatSng = 1;

    #endregion

    #region Misc

    /// <summary>End-of-function marker. Delphi: end_func</summary>
    public const string EndFunc = "}_end_func";

    /// <summary>No error state. Delphi: ERR_ST_NONE</summary>
    public const int ErrStNone = 0;

    /// <summary>Semi error state. Delphi: ERR_ST_SEME</summary>
    public const int ErrStSeme = 1;

    #endregion
}
